packages/jest-runtime/src/index.ts

```diff
@@ -302,6 +320,83 @@ class Runtime {
     return cliOptions;
   }
 
+  // unstable as it should be replaced by https://github.com/nodejs/modules/issues/393, and we don't want people to use it
+  unstable_shouldLoadAsEsm = Resolver.unstable_shouldLoadAsEsm;
+
+  private async loadEsmModule(
```

Comment:
Another limitation you did not mention seems to be that it does not use any of the logic we have for normal modules regarding mocks, isolation, etc., correct? If so, we could maybe leave a comment saying this needs to be rewritten completely to share code with `requireModule`

Reply:
mocks are not implemented no, but isolation is. THe runtime itself is per test, so isolation _should_ be correct. will need to implement mocks, though

Reply:
Sorry that was unclear, I meant `isolateModules`

Reply:
aha, gotcha. `isolateModules` doesn't work with ESM regardless since it either has to be static or async. We probably need to do something else clever